### PR TITLE
Attendance bug fixes and functionality

### PR DESCRIPTION
**Pull request by billyeh**
Comprising #471, attendance-switch-trainee branch, (and #457) and bower-reorg branch. With fixes to announcements, web requests, life studies, attendance, and meal seating

### CODE REVIEW

**Comment by jontey** [comment]
When editing a group leaveslip, the events are not selected on the calendar.

File: ap/attendance/views.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def get_context_data(self, **kwargs):
       event_id = self.request.POST.get('events')
       event = Event.objects.get(id=event_id)
       selected_date = event.date_for_week(int(selected_week))
+      event.date = selected_date
```
Comment (jontey) [inline ap/attendance/views.py]: GroupSlips where trainee is not in them but created it is still showing up. 
I used the account that i generated leaveslips for service scheduling and i got a bunch of group leaveslips on my page.

File: libraries/react/scripts/selectors/selectors.js [hunk 2]
```diff
@@ -132,15 +132,10 @@ export const getESRforWeek = createSelector(
...
-      status = categorizeEventStatus(esr.event)
-      if (status.roll || status.slip) {
-        esr.event.status = status
-        evs.push(esr)
-      }
-    });
+    let evs = esrs.map(esr => {
+      esr.event.status = categorizeEventStatus(esr.event)
+      return esr
+    })
```
Comment (jontey) [inline libraries/react/scripts/selectors/selectors.js]: What does getGroupSlipsforPeriod do? I dont think the trainee_names is used anywhere

File: ap/attendance/views.py [hunk 3]
```diff
@@ -100,6 +100,7 @@ def get_context_data(self, **kwargs):
       event_id = self.request.POST.get('events')
```
Comment (jontey) [inline ap/attendance/views.py]: Lin 59: The trainees selector is generating a lot of db calls. Perhaps consider prefetching groups

File: ap/attendance/views.py (hunk 1 above)
Comment (jontey) [inline ap/attendance/views.py]: Also when editing groupslips, the events are not selected automatically. the user has to select the events again before updating the leaveslip.

File: ap/schedules/views.py [hunk 4]
```diff
@@ -41,67 +40,12 @@ def get_queryset(self):
...
-          schedule.events.add(event)
-      else:
-        schedule = Schedule(trainee=trainee, term=event.term)
-        schedule.save()
-        schedule.events.add(event)
-
-    return super(EventUpdate, self).form_valid(form)
-
-
 class EventDelete(generic.DeleteView):
```
Comment (jontey) [inline ap/schedules/views.py]: This view is not used. Could you delete it?

File: ap/attendance/views.py (hunk 1 above)
Comment (jontey) [inline ap/attendance/views.py]: Ok. figured it out. If there is no event on the group slip calendar that falls within the time selected, then no events are selected. It's a small edge case which can be avoided by filling up all the empty slots on the calendar with something

**Comment by billyeh** [comment]
All right. Thanks for reviewing Jon and Leonard!

**Comment by billyeh** [comment]
@jontey Merging this before Saturday this week if there's nothing else to address :)

`multiselect.js` in meal seating static assets should also be moved to npm but https://www.npmjs.com/package/multiselect-two-sides is the package, and it has a very outdated jquery requirement that messes us up. We can address it later by either getting him to upgrade it or doing it ourselves.

**Pull request merged**